### PR TITLE
Code review - Update script.js

### PR DESCRIPTION
**Pull request by bedis-elacheche**

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -10,13 +10,14 @@ const winResults = [
...
-var clearButton = document.getElementById("clearButton");
-var infoP = document.getElementById("info");
+let isPlayer1;
+let board;
+let player1Status = [];
+let player2Status = [];
+const allSquares = document.getElementsByClassName("square");
+const startButton = document.getElementById("startButton");
+const clearButton = document.getElementById("clearButton");
+const infoP = document.getElementById("info");
```
Comment (bedis-elacheche) [inline script.js]: We're not reassigning those variables. We can use `const`

File: script.js [hunk 2]
```diff
@@ -27,8 +28,7 @@ function clearGame(){
     infoP.innerText = "Player1: X Player2: O";
     for (let i=0; i<allSquares.length;i++){
         allSquares[i].style.backgroundColor = "white";
-        allSquares[i].classList.remove("player1");
-        allSquares[i].classList.remove("player2");
+        allSquares[i].classList.remove("player1", "player2");
```
Comment (bedis-elacheche) [inline script.js]: remove accept many classnames, we can merge the two lines

File: script.js [hunk 3]
```diff
@@ -51,17 +51,17 @@ function play(button){
...
+        player2Status.push(id);
     }
     //update div with new class
     button.target.classList.add(player);
-    
+
     if (player1Status.length >= 3 || player2Status.length >= 3){
-        let win = checkWinner();
-        if (win["result"]){
+        const win = checkWinner();
```
Comment (bedis-elacheche) [inline script.js]: We're not reassigning win. We can use `const`

File: script.js [hunk 4]
```diff
@@ -51,17 +51,17 @@ function play(button){
...
+        if (win.result){
```
Comment (bedis-elacheche) [inline script.js]: When not using dynamic keys. Prefer using dot notation instead of brackets

File: script.js [hunk 5]
```diff
@@ -73,21 +73,22 @@ function play(button){
     //turn player
     isPlayer1 = !isPlayer1;
     player = isPlayer1 ? "player1" : "player2";
-    document.getElementById("info").innerText = symbols[player]+ " turn";
+    document.getElementById("info").innerText = `${symbols[player]} turn`;
```
Comment (bedis-elacheche) [inline script.js]: Use template literals when you have a mix of variables and strings

File: script.js [hunk 6]
```diff
@@ -73,21 +73,22 @@ function play(button){
...
 }
 
 function checkWinner(){
-    let playStatus = player1Status;
-    if (!isPlayer1){
-        playStatus = player2Status;
-    }
+    const playStatus = isPlayer1 ? player1Status : player2Status;
```
Comment (bedis-elacheche) [inline script.js]: Ternary operator: less code :)

File: script.js [hunk 7]
```diff
@@ -73,21 +73,22 @@ function play(button){
...
     for (const result of winResults){
-        let isWin = result.every(v=> playStatus.indexOf(v) > -1)
+        const isWin = result.every(v=> playStatus.indexOf(v) > -1)
         if (isWin) {
-            return {"result":true, "winner":isPlayer1, "line":result};
+            return {
+            	result: true,
+            	winner: isPlayer1,
+            	line:result
```
Comment (bedis-elacheche) [inline script.js]: "" can be ommited on object keys

File: script.js [hunk 8]
```diff
@@ -97,14 +98,14 @@ function removeEventListeners(){
     }
 }
 
-function gameOver(win){
+function gameOver({ line, winner }){
```
Comment (bedis-elacheche) [inline script.js]: we can destructure the object win, this will make us avoid typed `win.property` each time

File: script.js [hunk 9]
```diff
@@ -97,14 +98,14 @@ function removeEventListeners(){
...
     //show line
-    for (let index of win["line"]){
+    for (let index of line){
         document.getElementById(index).style.backgroundColor = "yellow";
     }
     //show winner message
-    let player = win["winner"] ? "player1" : "player2";
-    infoP.innerText = player + " Win!! click start button to continuer.";
+    const player = winner ? "player1" : "player2";
+    infoP.innerText = `${player} Win!! click start button to continue.`;
```
Comment (bedis-elacheche) [inline script.js]: Use template literals when you have a mix of variables and strings

File: script.js (hunk 1 above)
Comment (jingXuOne) [reply script.js]: so in es6, avoid use "var"?

Comment (bedis-elacheche) [reply script.js]: yes :)

**Pull request closed without merging**